MetaMask/metamask-extension
# E2e request queuing

## **Description**

Adds e2e tests for the request queuing system. Tests include enable the toggle settings, sending a tx from a different selected/global network which results in a switch chain request, and request queuing for multiple daps on different networks.



## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/1404

## **Manual testing steps**

1. `yarn && yarn build:test`
2. `yarn test:e2e:single test/e2e/tests/request-queuing/enable-queuing.spec.js --browser=chrome` 
3. `yarn test:e2e:single test/e2e/tests/request-queuing/switch-network.spec.js --browser=chrome`
4. `yarn test:e2e:single test/e2e/tests/request-queuing/multiple-networks-dapps-txs.spec.js --browser=chrome`
5. All request queuing tests should pass individually. `yarn test:e2e:chrome && yarn test:e2e:firefox` for a bonus.

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
